store/src/main/java/com/nytimes/android/external/store3/pipeline/RealMvRxStateStore.kt

```diff
@@ -0,0 +1,107 @@
+package com.airbnb.mvrx
+
+
+import com.nytimes.android.external.store3.pipeline.StateStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.channels.actor
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import java.util.*
+import kotlin.coroutines.CoroutineContext
+
+
+/**
+ * This is a container class around the actual state itself. It has a few optimizations to ensure
+ * safe usage of state.
+ *
+ * All state reducers are run in an actor to ensure that they don't have race
+ * conditions with each other.
+ *
+ */
+
+internal open class RealStateStore<S : Any>(initialState: S, final override val coroutineContext: CoroutineContext = Dispatchers.Default) : CoroutineScope, StateStore<S> {
```

Comment:
if this is really specific to MvRx, lets move it to a module. I don't think we should make mvrx a requirement to use this library.

Reply:
ignore for now